components/scream/src/share/io/output_manager.hpp

```diff
@@ -0,0 +1,179 @@
+  // First parse the param list for output.
+  // Starting with the stride.  NOTE: Only a stride of 1 is supported right now.  TODO: Allow for stride of any value.
+  Int stride = m_params.get<Int>("PIO Stride");
+  EKAT_REQUIRE_MSG(stride==1,"Error! Output only supports a PIO Stride of 1");  // TODO: Delete when no longer valid.
+  // Int comm_color = atm_comm.rank() % stride;
+  // auto pio_comm = atm_comm;  // TODO, EKAT should have a comm_split option (.split(comm_color));
+  // PIO requires a subsystem to begin.  TODO, the component coupler actually inits the subsystem for the ATM,
+  //                                     When the surface coupling is complete we can pass compid from the coupler
+  //                                     to the manager and switch the third arguement below from true to false.
+  // int compid=0;  // For CIME based builds this will be the integer ID assigned to the atm by the component coupler.  For testing we simply set to 0
+  MPI_Fint fcomm = MPI_Comm_c2f(pio_comm.mpi_comm());  // MPI communicator group used for I/O.  In our simple test we use MPI_COMM_WORLD, however a subset could be used.
+  eam_init_pio_subsystem(fcomm);   // Gather the initial PIO subsystem data creater by component coupler
+  
+  // Check for restart output
+  Int restart_hist_N = 0;
+  std::string restart_hist_OPTION = "None";
+  if (m_params.isSublist("Restart Control"))
+  {
+    auto& out_params = m_params.sublist("Restart Control");
+    make_restart_param_list(out_params);
+    new_output(out_params);
+    // Gather restart frequency info for restart history files
+    auto& freq_params = out_params.sublist("FREQUENCY");
+    restart_hist_N = freq_params.get<Int>("OUT_N");
+    restart_hist_OPTION = freq_params.get<std::string>("OUT_OPTION");
+  }
+  // Construct and store an output stream instance for each output request.
+  auto& list_of_files = m_params.get<std::vector<std::string>>("Output YAML Files");    // First grab the list of Output files from the control YAML
+  for (auto& it : list_of_files)
+  {
+    ekat::ParameterList out_params(it);
+    parse_yaml_file(it,out_params);
+    out_params.set<Int>("restart_hist_N",restart_hist_N);
+    out_params.set<std::string>("restart_hist_OPTION",restart_hist_OPTION);
+    new_output(out_params);
+  }
+}
+/*===============================================================================================*/
+// Overload run to allow for passing a TimeStamp or just directly a Real
+inline void OutputManager::run(util::TimeStamp& current_ts)
+{
+  if (no_output) { return; }
+  for (auto& it : m_output_streams)
+  {
+    it.run(*m_device_field_repo, *m_grids_manager, current_ts);
+  }
+}
+/*-----------------------------------------------------------------------------------------------*/
+inline void OutputManager::run(Real current_ts)
+{
+  if (no_output) { return; }
+  for (auto& it : m_output_streams)
+  {
+    it.run(*m_device_field_repo, *m_grids_manager, current_ts);
```

Comment:
Should I do the same for the field repo?

Reply:
If each `output_type` is used only by this OutputManager and not shared amongst any other entities (which is probably true), then it's safe and convenient to pass them the grids manager and the field repo to use. If they were shared among different entities/output managers/whatever, it would make more sense to pass the grids manager and the field repo as arguments to their `run` methods, since an output_type would be `run`ning for more than one entity in that case.